applications/dashboard/controllers/Api/TokensApiController.php

```diff
@@ -0,0 +1,316 @@
+                'dateInserted:dt' => 'When the token was generated.'
+            ], 'token');
+        }
+        return $this->fullSchema;
+    }
+
+    /**
+     * Get a signed copy of a token.
+     *
+     * @param int $id
+     * @param array $query
+     * @throws NotFoundException if this is not an active token.
+     * @return array
+     */
+    public function get($id, array $query) {
+        $this->permission('Garden.Tokens.Add');
+
+        $in = $this->schema([
+            'id',
+            'transientKey:s' => 'A valid CSRF token for the current user.'
+        ], 'in')->add($this->idSchema())->setDescription('Reveal a usable authentication token.');
+        $out = $this->schema($this->sensitiveSchema(), 'out');
+
+        $query['id'] = $id;
+        $query = $in->validate($query);
+        $this->validateTransientKey($query['transientKey']);
+
+        $row = $this->token($id);
+        if ($row['UserID'] != $this->session->UserID) {
+            if ($this->session->checkPermission('Garden.Settings.Manage') === false) {
+                throw new NotFoundException('Access Token');
+            }
+        }
+        $this->isActiveToken($row, true);
+        $this->prepareRow($row);
+
+        $result = $out->validate($row);
+        return $result;
+    }
+
+    /**
+     * Get an ID-only schema for token records.
+     *
+     * @return Schema
+     */
+    public function idSchema() {
+        if (!isset($this->idSchema)) {
+            $this->idSchema = $this->schema(
+                ['id:i' => 'The numeric ID of a token.'],
+                'tokenID'
+            );
+        }
+        return $this->idSchema;
+    }
+
+    /**
+     * List active tokens for the current user.
+     *
+     * @return array
+     */
+    public function index() {
+        $this->permission('Garden.Tokens.Add');
+
+        $in = $this->schema([], 'in');
+        // Full access token details are not available in the index. Use GET on a single ID for sensitive information.
+        $out = $this->schema([
+            ':a' => $this->schema([
+                'accessTokenID',
+                'name',
+                'dateInserted'
+            ])->add($this->fullSchema())
+        ], 'out')->setDescription('Get a list of authentication token IDs for the current user.');
+
+        $rows = $this->accessTokenModel->getWhere([
+            'UserID' => $this->session->UserID,
+            'Type' => self::TOKEN_TYPE
+        ], '', 'asc', self::RESPONSE_LIMIT)->resultArray();
+        $activeTokens = [];
+        foreach ($rows as $token) {
+            if ($this->isActiveToken($token) === false) {
+                continue;
+            }
+            $activeTokens[] = $token;
+        }
+        unset($token);
+        array_walk($activeTokens, [$this, 'prepareRow']);
+
+        $result = $out->validate($activeTokens);
+        return $result;
+    }
+
+    /**
+     * Issue a new transient key for the current user.
```

Comment:
access token?

We need to make sure to only use one terminology when speaking about something.

`->setDescription('Issue a new authentication token for the current user.');`
should probably by
`->setDescription('Issue a new access token for the current user.');`